sklearn/linear_model/tests/test_least_angle.py

```diff
@@ -117,20 +117,16 @@ def test_all_precomputed():
             assert_array_almost_equal(expected, got)
 
 
-@pytest.mark.filterwarnings("ignore: `rcond` parameter will change")
-# numpy deprecation
 def test_lars_lstsq():
     # Test that Lars gives least square solution at the end
     # of the path
     X1 = 3 * X  # use un-normalized dataset
     clf = linear_model.LassoLars(alpha=0.0)
     clf.fit(X1, y)
-    coef_lstsq = np.linalg.lstsq(X1, y, rcond=None)[0]
+    coef_lstsq = np.linalg.lstsq(X1, y)[0]
```

Comment:
Interesting this is still a warning for numpy<2, so does that mean that the only builds we are turning warnings into errors have numpy>=2? The warning was introduced a while ago numpy 1.14 according to the [Numyp 1.26 doc](https://numpy.org/doc/1.26/reference/generated/numpy.linalg.lstsq.html)

For example this snippet with numpy 1.26.4 gives a warning:
```py
import numpy as np

x = np.array([0, 1, 2, 3])
y = np.array([-1, 0.2, 0.9, 2.1])
A = np.vstack([x, np.ones(len(x))]).T

np.linalg.lstsq(A, y)
```

```
<ipython-input-7-3594d4caf658>:1: FutureWarning: `rcond` parameter will change to the default of machine precision times ``max(M, N)`` where M and N are the input matrix dimensions.
To use the future default and silence this warning we advise to pass `rcond=None`, to keep using the old, explicitly pass `rcond=-1`.
  m, c = np.linalg.lstsq(A, y)[0]
```

Reply:
Not sure what is going on, but this seems weird, there are some builds with `numpy<2` that have `SKLEARN_WARNINGS_AS_ERRORS: '1'` e.g. `pymin_conda_forge_openblas_ubuntu_2204` which has numpy 1.21.6.

Using its lockfile locally I do get an error:
```
SKLEARN_WARNINGS_AS_ERRORS=1 pytest sklearn/linear_model/tests/test_least_angle.py -k test_lars_lstsq
```

Reply:
Indeed that's weird. I put back the warning filter and added a TODO comment to remove it when it's time. I also removed setting rcond=None because it's either we set it to None to disable the warning or we left it to the default and filter the warning.

Reply:
I forgot to say, but basically this makes me a bit worried that somehow there is something wrong with our warnings as errors setup or maybe some weird side-effect between the tests.

Just off the top of my head, an imaginary scenario that could be far from reality: you could imagine that there is one test that ignores the warning and that since it was a warning 'once' (the default filter IIRC) other tests that run afterwards don't emit the same warnings.

In this scenario: I could easily imagine that the tests run in the CI pass fine, but that if you were to run individual tests with `SKLEARN_WARNINGS_AS_ERRORS=1` it would fail.

At the same time, I think we should encourage this kind of cleaning, so maybe this should be a "let's try this and fix it later if the need arises".

Reply:
To double-check would I ran the test for each separated file and checked that there is no error so at least if there are some side effects they don't happen across files

```zsh
files=$(git d upstream/main... --name-only | sort | uniq) 
# ${=files} needed in zsh, I think use $files in bash?
for f in ${=files}; do SKLEARN_WARNINGS_AS_ERRORS=1 pytest $f; done
```

Reply:
Thanks for checking !

> At the same time, I think we should encourage this kind of cleaning, so maybe this should be a "let's try this and fix it later if the need arises".

:+1:
That's what I'm thinking in this situation because it's only about our CI. And it's not like I'm adding filters to hide stuff but removing them instead so the worst that could happen is that somehow a test does not emit a warning because of a side effect as you suggest. It could already be the case for all tests that didn't have a filter in the first place :smile:

Reply:
Let's merge this and see what happens.